prow/git/git.go

```diff
@@ -74,6 +75,11 @@ func (c *Client) Clean() error {
 // NewClient returns a client that talks to GitHub. It will fail if git is not
 // in the PATH.
 func NewClient() (*Client, error) {
```

Comment:
These changes should be separate from the GitHub OAuth ones

Reply:
I'm uncertain how do you want to me proceed, create another PR or you will take care of this in #14609?

Reply:
Just separate PRs. Let this first PR handle the OAuth bits, follow up later with `git`.

Reply:
So I made it minimal to just about consuming the host configuration.